mantidproject/mantid
# Set parameter visibility for InstrumentViewer

ILL users found that certain parameters stored in IPFs are very verbose, with multiple lines of text, and that has a severe impact on user experience using the Pick tab in the InstrumentViewer, namely the picking slows down significantly. The proposed solution is to add an attribute to certain parameters stored in IPFs defining whether they should be visible from the InstrumentViewer. This attribute is by default set to 'true'.

**To test:**

1. Load some ILL data, e.g. 093409.nxs of D33 from UniTest/ILL/D33
2. Open the InstrumentViewer and go to the Pick tab, hover over a sensitive element of the instrument.
3. There should be only a handful of parameters printed from D33 component-link (marked as 'Parameters from: D33'), and none longer than one line.
4. Go to the instrument/D33_Parameters.xml and change the 'visible' attribute value from 'false' to 'true' of one of the non-visible parameters, e.g. 'logbook_default_parameters'.
5. Re-load D33 data, and go again to the Pick tab of the InstrumentViewer.
6. The parameters contained in the chosen parameter, e.g. 'logbook_default_parameters', should now be printed.

Part of  #31052  


Release notes added.



---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
